common/analysis/lint_rule_status.cc

```diff
@@ -105,6 +108,36 @@ struct LintViolationWithStatus {
   }
 };
 
+std::string LintStatusFormatter::ReplaceWithHelperTokens(
+    const std::vector<verible::TokenInfo>& tokens, absl::string_view reason,
+    absl::string_view path, absl::string_view base) const {
+  if (tokens.empty()) {
+    return std::string(reason);
+  }
+
+  size_t end_pos = reason.find("@", 0), beg_pos = 0;
+  std::ostringstream s;
+  for (const auto& token : tokens) {
+    if (end_pos == std::string_view::npos) {
+      s << reason.substr(beg_pos);
+      break;
+    }
+
+    if (!(end_pos != 0 && reason[end_pos - 1] == '\\')) {
+      s << reason.substr(beg_pos, end_pos - beg_pos);
+      s << path << ":";
```

Comment:
We should probably write in the header documentation of this function that the tokens all must come from the same filename (right now, this is always the case we look file by file, but we might parse multiple files and thus error messages that span over multiple files later)

Reply:
Done